crates/teloxide-core/src/types/rgb.rs

```diff
@@ -0,0 +1,96 @@
+use rgb::RGB8;
+use serde::{de::Visitor, Deserialize, Serialize};
+
+/// RGB color format
+#[derive(Clone, Copy, Debug)]
+#[derive(PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct Rgb {
+    pub r: u8,
+    pub g: u8,
+    pub b: u8,
+}
+
+impl Rgb {
+    fn to_u32(self) -> u32 {
+        u32::from_be_bytes([0, self.r, self.g, self.b])
+    }
+
+    fn from_u32(rgb: u32) -> Self {
+        let [_, r, g, b] = rgb.to_be_bytes();
+        Rgb { r, g, b }
+    }
+}
+
+impl Serialize for Rgb {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_u32(self.to_u32())
+    }
+}
+
+impl<'de> Deserialize<'de> for Rgb {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        struct V;
+
+        impl Visitor<'_> for V {
+            type Value = Rgb;
+
+            fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+                formatter.write_str("an integer represeting an RGB color")
+            }
+
+            fn visit_u32<E>(self, v: u32) -> Result<Self::Value, E>
+            where
+                E: serde::de::Error,
+            {
+                Ok(Self::Value::from_u32(v))
+            }
+
+            fn visit_u64<E>(self, v: u64) -> Result<Self::Value, E>
+            where
+                E: serde::de::Error,
+            {
+                self.visit_u32(v.try_into().map_err(|_| E::custom("rgb value doesn't fit u32"))?)
+            }
+        }
+
+        deserializer.deserialize_u32(V)
+    }
+}
+
+impl From<RGB8> for Rgb {
+    fn from(color: RGB8) -> Self {
+        Rgb { r: color.r, g: color.g, b: color.b }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use serde::{Deserialize, Serialize};
+
+    use super::*;
+
+    #[test]
+    fn bytes() {
+        assert_eq!(Rgb { r: 0xAA, g: 0xBB, b: 0xCC }.to_u32(), 0x00AABBCC);
+        assert_eq!(Rgb::from_u32(0x00AABBCC), Rgb { r: 0xAA, g: 0xBB, b: 0xCC });
+    }
+
+    #[test]
+    fn json() {
+        #[derive(Debug, PartialEq, Eq, Serialize, Deserialize)]
+        struct Struct {
+            color: Rgb,
+        }
+
+        let json = format!(r#"{{"color":{}}}"#, 0x00AABBCC);
```

Comment:
Why nested curly brackets?

Reply:
I think I just copypasted old teloxide code, I will fix it

Reply:
It turns out that they are for escaping purposes here.

I get rustc error without them:
```
invalid format string: expected `'}'`, found `'"'`
if you intended to print `{`, you can escape it using `{{`
```